File: SmartPay/Models/CheckProduct.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartPay.Models;

public class CheckProduct
{
    public int CheckUid { get; set; }
    public Check Check { get; set; }
    
    public int ProductId { get; set; }
    public Product Product { get; set; }
    
    public int Cost { get; set; }
    public int Amount { get; set; }
}

File: SmartPay/ClientApp/src/api/models/CheckProduct.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { Check } from './Check';
import type { Product } from './Product';

export type CheckProduct = {
    checkUid?: number;
    check?: Check;
    productId?: number;
    product?: Product;
    cost?: number;
    amount?: number;
};